docs/source/install/ewc_ha.rst

```diff
@@ -0,0 +1,277 @@
+
+st2timersengine
+_______________
+st2timersengine is responsible for scheduling all user specified `timers <https://docs.stackstorm.com/rules.html#timers>`_ aka st2 cron.
+Only single replica is created via K8s Deployment as timersengine can't work in active-active mode at the moment
+(multiple timers will produce duplicated events) and it relies on K8s failover/reschedule capabilities to address cases of process failure.
+
+st2workflowengine
+_________________
+st2workflowengine drives the execution of orquesta workflows and actually schedules actions to run by another component ``st2actionrunner``.
+Multiple st2workflowengine processes can run in active-active mode and so minimum ``2`` K8s Deployment replicas are created by default.
+All the workflow engine processes will share the load and pick up more work if one or more of the processes become available.
+
+.. note::
+  As Mistral is going to be deprecated and removed from StackStorm platform soon, Helm chart relies only on
+  :doc:`Orquesta st2workflowengine </orquesta/index>` as a new native workflow engine.
+
+st2notifier
+___________
+Multiple st2notifier processes can run in active-active mode, using connections to RabbitMQ and MongoDB and generating triggers based on
+action execution completion as well as doing action rescheduling.
+In an HA deployment minimum ``2`` replicas of st2notifier is running, requiring coordination backend, which is ``etcd`` in our case.
+
+st2sensorcontainer
+__________________
+st2sensorcontainer manages StackStorm sensors: starts, stops and restarts them as a subprocesses.
+At the moment K8s configuration consists of Deployment with hardcoded ``1`` replica.
+Future plans are to re-work this setup and benefit from Docker-friendly `single-sensor-per-container mode #4179 <https://github.com/StackStorm/st2/pull/4179>`_
+(since st2 ``v2.9``) as a way of :doc:`/reference/sensor_partitioning`, distributing the computing load
+between many pods and relying on K8s failover/reschedule mechanisms, instead of running everything on ``1`` single instance of st2sensorcontainer.
+
+st2actionrunner
+_______________
+Stackstorm workers that actually execute actions.
+``5`` replicas for K8s Deployment are configured by default to increase StackStorm ability to execute actions without excessive queuing.
+Relies on ``etcd`` for coordination. This is likely the first thing to lift if you have a lot of actions
+to execute per time period in your StackStorm cluster.
+
+st2garbagecollector
+___________________
+Service that cleans up old executions and other operations data based on setup configurations.
+Having ``1`` st2garbagecollector replica for K8s Deployment is enough, considering its periodic execution nature.
+By default this process does nothing and needs to be configured in st2.conf settings (via ``values.yaml``).
+Purging stale data can significantly improve cluster abilities to perform faster and so it's recommended to configure st2garbagecollector in production.
+
```

Comment:
Worth mentioning docker-registry and kube-registry-proxy here? They're optional services... based entirely on st2.packs config.

Reply:
Per PR comment I'm still waiting for https://github.com/StackStorm/k8s-st2/pull/78 to be finished.

Just took some descriptions from that PR and added new section in https://github.com/StackStorm/st2docs/pull/795/commits/806eebb0af92519807b121e29c8aa134bfd792a9 for private docker registry.